File: SingletonJava/src/singletonjava/SingletonObject.java

package singletonjava;

/**
 *
 * @author starb
 */
public class SingletonObject {
    
    private static SingletonObject instance = null;
    
    private SingletonObject()
    {
        System.out.println("Llamando al constructor de SingletonObject");
    }
    
    public static SingletonObject getInstance()
    {
        System.out.println("Llamando a getInstance de SingletonObject");
        
        if (instance == null) {
            instance = new SingletonObject();
        }
        
        return instance;
    }
    
    public void doSomething()
    {
        System.out.println("Llamando a doSomething de SingletonObject");
    }
}

File: Singleton/Singleton/SingletonObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Singleton
{
    class SingletonObject
    {
        private static SingletonObject instance = null;
        public static SingletonObject Instance
        {
            get
            {
                Console.WriteLine("Llamando a GetInstance de SingletonObject");

                if (instance == null)
                {
                    instance = new SingletonObject();
                }

                return instance;
            }
        }

        private SingletonObject()
        {
            Console.WriteLine("Llamando al constructor de SingletonObject");
        }

        public void DoSomething()
        {
            Console.WriteLine("Llamando a DoSomething de SingletonObject");
        }
    }
}
